src/routes/PrivateRoute.tsx

```diff
@@ -0,0 +1,18 @@
+import { useAppSelector } from 'app/hooks';
+import React from 'react';
+import { Navigate, useLocation } from 'react-router-dom';
+
+type PrivateRouteType = {
+  children: React.ReactNode;
+};
+
+export function PrivateRoute({ children }: PrivateRouteType): any {
+  let location = useLocation();
+  const { user } = useAppSelector((state) => state.user);
+
+  if (user?.authToken) {
+    return children;
+  }
+
+  return <Navigate to="/login" state={{ from: location }} replace />;
```

Comment:
clarification: is it location or you have to use location.state for redirect

Reply:
Yes we can use that as well. But this location object has other properties like pathname and state and few other queries . So I need this in my login page . So that i can redirect back to the same page after successfull login.